src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java

```diff
@@ -358,28 +369,71 @@ public void validate() {
 
     /**
      * Does this feed version have any critical errors that would prevent it being loaded to OTP?
-     * @return whether feed version has critical errors
+     * @return whether the feed version has any critical errors
      */
     public boolean hasCriticalErrors() {
-        return hasCriticalErrorsExceptingDate() ||
-                validationResult.lastCalendarDate == null ||
-                (LocalDate.now()).isAfter(validationResult.lastCalendarDate);
+        return hasValidationAndLoadErrors() ||
+            hasFeedVersionExpired() ||
+            hasHighSeverityErrorTypes();
     }
 
     /**
-     * Does this feed have any critical errors other than possibly being expired?
-     * @return whether feed version has critical errors (outside of expiration)
+     * Does this feed have any validation or load errors?
+     * @return whether feed version has any validation or load errors
      */
-    private boolean hasCriticalErrorsExceptingDate() {
+    private boolean hasValidationAndLoadErrors() {
         if (validationResult == null)
             return true;
-
         return validationResult.fatalException != null ||
             !validationSummary().bounds.areValid() ||
             feedLoadResult.stopTimes.rowCount == 0 ||
             feedLoadResult.trips.rowCount == 0 ||
             feedLoadResult.agency.rowCount == 0;
+    }
+
+    /**
+     * Has this feed expired?
+     * @return whether feed version is outside of expiration
```

Comment:
Can you update this to also say it'll return true if the `validationResult.lastCalendar` value is null?

Reply:
Updated.

Reply:
Ok, but can it also be added to this specific method?

Reply:
Updated the correct method!